server/src/main/java/org/elasticsearch/common/util/set/Sets.java

```diff
@@ -148,63 +143,12 @@ public static <T> Collector<T, SortedSet<T>, SortedSet<T>> toSortedSet() {
      * @return an unmodifiable set where the underlying set is sorted
      */
     public static <T> Collector<T, SortedSet<T>, SortedSet<T>> toUnmodifiableSortedSet() {
-        return new UnmodifiableSortedSetCollector<>();
+        return Collector.of(TreeSet::new, SortedSet::add, Sets::addAllMutable, Collections::unmodifiableSortedSet);
     }
 
-    abstract static class AbstractSortedSetCollector<T> implements Collector<T, SortedSet<T>, SortedSet<T>> {
-
-        @Override
-        public Supplier<SortedSet<T>> supplier() {
-            return TreeSet::new;
-        }
-
-        @Override
-        public BiConsumer<SortedSet<T>, T> accumulator() {
-            return SortedSet::add;
-        }
-
-        @Override
-        public BinaryOperator<SortedSet<T>> combiner() {
-            return (s, t) -> {
-                s.addAll(t);
-                return s;
-            };
-        }
-
-        public abstract Function<SortedSet<T>, SortedSet<T>> finisher();
-
-        public abstract Set<Characteristics> characteristics();
-
-    }
-
-    private static class SortedSetCollector<T> extends AbstractSortedSetCollector<T> {
-
-        @Override
-        public Function<SortedSet<T>, SortedSet<T>> finisher() {
-            return Function.identity();
-        }
-
-        static final Set<Characteristics> CHARACTERISTICS = Collections.unmodifiableSet(EnumSet.of(Characteristics.IDENTITY_FINISH));
-
-        @Override
-        public Set<Characteristics> characteristics() {
-            return CHARACTERISTICS;
-        }
-
-    }
-
-    private static class UnmodifiableSortedSetCollector<T> extends AbstractSortedSetCollector<T> {
-
-        @Override
-        public Function<SortedSet<T>, SortedSet<T>> finisher() {
-            return Collections::unmodifiableSortedSet;
-        }
-
-        @Override
-        public Set<Characteristics> characteristics() {
-            return Collections.emptySet();
-        }
-
+    private static <T, S extends Set<T>> S addAllMutable(S a, S b) {
```

Comment:
I'm late for the review, but the combiner function should generate a new set combined from `a` and `b`. It exists for the case if a stream is parallel and the end result needs to be merged from sub-results produced by different threads.

Reply:
I replicated this behavior from the previous implementation, possibly this collector was never used in a parallel stream?

I checked a javadoc for `java.util.stream.Collector#combiner` and it states:
```
A function that accepts two partial results and merges them. The combiner function may fold state from one argument into the other and return that, or may return a new result container.
```
Is there a clarification somewhere else for a concurrent case?